processor/src/lib.rs

```diff
@@ -122,36 +139,65 @@ pub fn execute_iter(program: &Program, inputs: &ProgramInputs) -> VmStateIterato
 // PROCESS
 // ================================================================================================
 
-pub struct Process {
+pub struct Process<ADV>
```

Comment:
nit: I'd prefer to use just `A` for type parameter name instead of `ADV`. I think we almost never use multiple letters for type parameter names - so, this would be more consistent.

Reply:
Done! This is indeed an unspoken rule for the general community as everyone uses a single char. It can get a bit confusing if we have multiple generics, but that is something we generally avoid anyway.